lib/puppet/provider/user/directoryservice.rb

```diff
@@ -151,6 +151,10 @@ def self.generate_attribute_hash(input_hash)
         attribute_hash[:iterations] = get_salted_sha512_pbkdf2('iterations', embedded_binary_plist)
       elsif embedded_binary_plist['SALTED-SHA512']
         attribute_hash[:password] = get_salted_sha512(embedded_binary_plist)
+      else
```

Comment:
This is a bit funny because we have a catchall conditional but we're specifically calling the `get_salted_sha512_pbkdf2` function, which seems specific for SALTED-SHA512-PBKDF2. Why does this work, and what conditionals are we catching or what algorithm are we using?